packages/opentelemetry-resources/src/platform/node/ProcessDetectorSync.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { Resource } from '../../Resource';
+import { DetectorSync, ResourceAttributes } from '../../types';
+import { ResourceDetectionConfig } from '../../config';
+import { IResource } from '../../IResource';
+import * as os from 'os';
+
+/**
+ * ProcessDetectorSync will be used to detect the resources related current process running
+ * and being instrumented from the NodeJS Process module.
+ */
+class ProcessDetectorSync implements DetectorSync {
+  detect(_config?: ResourceDetectionConfig): IResource {
+    const attributes: ResourceAttributes = {
+      [SemanticResourceAttributes.PROCESS_PID]: process.pid,
+      [SemanticResourceAttributes.PROCESS_EXECUTABLE_NAME]: process.title,
+      [SemanticResourceAttributes.PROCESS_EXECUTABLE_PATH]: process.execPath,
+      [SemanticResourceAttributes.PROCESS_COMMAND_LINE]: process.argv.join(' '),
```

Comment:
This is likely to be misleading if one of the argv items contains white spaces.

Reply:
👍 

constructing the exact command line is likely not possible. One option is to wrap each argument in quotes (escaping quotes in the argument itself). Another would be to escape whitespace characters.

Wrapping in quotes while escaping at the same time can be done quite easily with JSON.stringify. The JSON escaping rules match the shell escaping rules close enough for most purposes:

```suggestion
      [SemanticResourceAttributes.PROCESS_COMMAND_LINE]: process.argv.map(JSON.stringify).join(' '),
```

We are also missing the results of [`process.execArgv`](https://nodejs.org/api/process.html#processexecargv).

Reply:
sounds reasonable

Reply:
I can go ahead and remove `process.command_line` as it's redundant, and the spec prefers `process.command_args` anyhow.